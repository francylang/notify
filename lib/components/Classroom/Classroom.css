.student-classroom-container {
  display: flex;
  width: 100%;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center; }

.student-classroom {
  width: 95%;
  display: flex;
  flex-wrap: wrap;
  align-items: left; }

.teacher-name {
  margin: 215px 0 18px 55px;
  color: #E3C574;
  font-size: 30px;
  font-weight: lighter; }

.drop-down {
  width: 20%;
  background-color: transparent;
  font-size: 20px;
  font-weight: lighter;
  margin: 212px 111px 16px 0px;
  width: 348px;
  color: #777;
  font-weight: 300;
  background-color: #fff;
  box-shadow: 1px 2px 10px -2px rgba(0, 0, 0, 0.3);
  border-radius: 3px;
  transition: all 375ms ease-in-out; }

option {
  font-family: sans-serif; }

.drop-down:focus {
  outline-color: #1a90bf; }

.class-drop-down {
  display: flex;
  flex-direction: row;
  justify-content: space-between; }

@media screen and (max-width: 320) {
  .student-classroom {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-wrap: wrap; } }

@media screen and (max-width: 650px) {
  .student-classroom {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-wrap: wrap;
    margin: 0px 10px 0 10px; }
  .teacher-name {
    font-size: 20px;
    margin: 211px 0 16px 105px; } }

@media screen and (max-width: 720px) {
  .teacher-name {
    font-size: 20px;
    margin: 211px 0 16px 87px; } }

@media screen and (min-width: 2000px) {
  .teacher-name {
    margin: 215px 0 18px 100px; } }
