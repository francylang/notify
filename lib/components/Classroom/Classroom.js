import React from 'react';
import PropTypes from 'prop-types';
import StudentLink from '../StudentLink/StudentLink.js';

const Classroom = ({teacher, students, deleteStudent}) => {
  return (
    <div>
      <div className='class-drop-down'>
        <h2 className='teacher-name'>{teacher}'s Classroom</h2>
      </div>
      <div className='student-classroom'>
      {
        students.map(({ name, id }) => {
          return <StudentLink
            key={ name }
            id={ id }
            name={ name }
            deleteStudent={ () => deleteStudent(id) }
          />;
        })
      }
      </div>
    </div>
  );
};

Classroom.propTypes = {
  teacher: PropTypes.string,
  students: PropTypes.array,
  deleteStudent: PropTypes.func
};

export default Classroom;
