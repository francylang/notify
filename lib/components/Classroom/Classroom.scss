$main-color: #1a90bf ;
$main-color-hover: #405780;

.student-classroom {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
}

.teacher-name {
  margin: 240px 0 18px 59px;
  color: #808080;
  font-size: 30px;
  font-weight: lighter;
}

.drop-down {
  width: 20%;
  background-color: transparent;
  font-size: 20px;
  font-weight: lighter;
  margin: 212px 111px 16px 0px;
  width: 348px;
  color: #777;
  font-weight: 300;
  background-color: #fff;
  box-shadow: 1px 2px 10px -2px rgba(0,0,0,0.3);
  border-radius: 3px;
  transition: all 375ms ease-in-out;
}

option {
  font-family: sans-serif;
}

.drop-down:focus {
  outline-color: $main-color;
}

.class-drop-down {
  display: flex;
  flex-direction: row;
  justify-content: space-between;

}

@media screen and (max-width: 320) {
  .student-classroom {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-wrap: wrap;
  }
}

@media screen and (max-width: 650px) {
  .student-classroom {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-wrap: wrap;
    margin: 0px 10px 0 10px;
  }

  .teacher-name {
    font-size: 20px;
    margin: 211px 0 16px 105px;
  }

@media screen and (max-width: 720px) {

  .teacher-name {
    font-size: 20px;
    margin: 211px 0 16px 87px;
  }
