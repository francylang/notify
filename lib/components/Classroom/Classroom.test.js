import React from "react";
import Classroom from './Classroom';
import { shallow, mount } from "enzyme";
import { MemoryRouter } from 'react-router-dom';


describe('CLASSROOM', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Classroom students={[{ name: 'Francy Lang', parentContact: 'Pam Lang', phone: '5029052136'}]}/>);
  });

  it('should exist', () => {

    expect(wrapper).toBeDefined();
  });

  it('should match the Classroom snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });
});

describe('classroom of teacher and students', () => {

  it.skip('should receive students and teacher', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Classroom students={[{ name: 'Francy Lang', parentContact: 'Pam Lang', phone: '5029052136'}]}
                   teacher={'Ms. Lang'}
                 />
      </MemoryRouter>
    );

    expect(wrapper.find('.teacher-name').text()).toEqual("Ms. Lang's Classroom");
    expect(wrapper.find('StudentLink').length).toEqual(1);
  });

  it.skip('should delete a student', () => {
    const mockDelete = jest.fn();
    const wrapper = mount(
      <MemoryRouter>
        <Classroom students={[{ id: 5, name: 'Francy Lang', parentContact: 'Pam Lang', phone: '5029052136'}]}
                   teacher={'Ms. Lang'}
                   deleteStudent={mockDelete}
                 />
      </MemoryRouter>
    );
    const deleteButton = wrapper.find('.delete-student-btn');
    deleteButton.simulate('click');
    expect(mockDelete.mock.calls.length).toEqual(1);
  });
});
