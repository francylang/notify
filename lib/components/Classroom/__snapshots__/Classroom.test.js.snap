// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLASSROOM should match the Classroom snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
        className="teacher-and-summary"
    >
        <h2
            className="teacher-name"
        >
            's Classroom
        </h2>
    </div>
    <div
        className="student-classroom-container"
    >
        <div
            className="student-classroom"
        >
            <StudentLink
                deleteStudent={[Function]}
                id={undefined}
                name="Francy Lang"
                studentInfo={
                    Array [
                        Object {
                          "name": "Francy Lang",
                          "parentContact": "Pam Lang",
                          "phone": "5029052136",
                        },
                      ]
                }
                studentValue={undefined}
            />
        </div>
    </div>
</div>,
  "nodes": Array [
    <div>
      <div
            className="teacher-and-summary"
      >
            <h2
                  className="teacher-name"
            >
                  's Classroom
            </h2>
      </div>
      <div
            className="student-classroom-container"
      >
            <div
                  className="student-classroom"
            >
                  <StudentLink
                        deleteStudent={[Function]}
                        id={undefined}
                        name="Francy Lang"
                        studentInfo={
                              Array [
                                    Object {
                                      "name": "Francy Lang",
                                      "parentContact": "Pam Lang",
                                      "phone": "5029052136",
                                    },
                                  ]
                        }
                        studentValue={undefined}
                  />
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Classroom
        students={
                Array [
                        Object {
                          "name": "Francy Lang",
                          "parentContact": "Pam Lang",
                          "phone": "5029052136",
                        },
                      ]
        }
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "students": Array [
            Object {
              "name": "Francy Lang",
              "parentContact": "Pam Lang",
              "phone": "5029052136",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
                    className="teacher-and-summary"
          >
                    <h2
                              className="teacher-name"
                    >
                              's Classroom
                    </h2>
          </div>
          <div
                    className="student-classroom-container"
          >
                    <div
                              className="student-classroom"
                    >
                              <StudentLink
                                        deleteStudent={[Function]}
                                        id={undefined}
                                        name="Francy Lang"
                                        studentInfo={
                                                  Array [
                                                            Object {
                                                              "name": "Francy Lang",
                                                              "parentContact": "Pam Lang",
                                                              "phone": "5029052136",
                                                            },
                                                          ]
                                        }
                                        studentValue={undefined}
                              />
                    </div>
          </div>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div>
          <div
                    className="teacher-and-summary"
          >
                    <h2
                              className="teacher-name"
                    >
                              's Classroom
                    </h2>
          </div>
          <div
                    className="student-classroom-container"
          >
                    <div
                              className="student-classroom"
                    >
                              <StudentLink
                                        deleteStudent={[Function]}
                                        id={undefined}
                                        name="Francy Lang"
                                        studentInfo={
                                                  Array [
                                                            Object {
                                                              "name": "Francy Lang",
                                                              "parentContact": "Pam Lang",
                                                              "phone": "5029052136",
                                                            },
                                                          ]
                                        }
                                        studentValue={undefined}
                              />
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Classroom
    students={
        Array [
            Object {
              "name": "Francy Lang",
              "parentContact": "Pam Lang",
              "phone": "5029052136",
            },
          ]
    }
/>,
}
`;
